### PR TITLE
Add successful query feature

### PR DESCRIPTION
**Pull request by Sushisource**
Added query success feature

part of https://github.com/temporalio/sdk-features/issues/82

(todo: Go not erroring right)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,9 @@ Prerequisites:
 * [Go](https://golang.org/) 1.17+
 * [JDK](https://adoptium.net/?variant=openjdk11&jvmVariant=hotspot) 11+
 * [Node](https://nodejs.org) 16+
+* [Python](https://www.python.org/) 3.7+
```
Comment (cretz) [inline README.md:27]: I think we can change this to 3.10+ if that's ok with everyone

File: features/query/successful_query/feature.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	"go.temporal.io/sdk-features/harness/go/harness"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/workflow"
+)
+
+var Feature = harness.Feature{
+	Workflows: Workflow,
+	CheckResult: func(ctx context.Context, r *harness.Runner, run client.WorkflowRun) error {
+		q1, err := r.Client.QueryWorkflow(ctx, run.GetID(), run.GetRunID(), "counterQ")
+		r.Assert.Nil(err)
```
Comment (cretz) [inline features/query/successful_query/feature.go:14]: `r.Assert.NoError` is better

File: features/query/successful_query/feature.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		r.Assert.Equal(1, q2)
+		err = r.Client.SignalWorkflow(ctx, run.GetID(), run.GetRunID(), "counterInc", nil)
+		r.Assert.Nil(err)
+		err = r.Client.SignalWorkflow(ctx, run.GetID(), run.GetRunID(), "counterInc", nil)
+		r.Assert.Nil(err)
+		err = r.Client.SignalWorkflow(ctx, run.GetID(), run.GetRunID(), "counterInc", nil)
+		r.Assert.Nil(err)
+		q3, err := r.Client.QueryWorkflow(ctx, run.GetID(), run.GetRunID(), "counterQ")
+		r.Assert.Nil(err)
+		// TODO: Something isn't right with go runner. This should fail and it does not
```
Comment (cretz) [inline features/query/successful_query/feature.go:29]: Ah, so this is a bit weird with how the `Assertion` library works. You have to:

```go
if !r.Assert.Equal(50, q3) {
  return r.LastAssertErr
}
```

Alternatively you could use `r.Require.Equal(50, q3)` which panics

File: harness/go/harness/runner.go [hunk 4]
```diff
@@ -285,11 +285,11 @@ func (r *Runner) QueryUntilEventually(
 
 // Close closes this runner.
 func (r *Runner) Close() {
-	if r.Worker == nil {
+	if r.Worker != nil {
```
Comment (cretz) [inline harness/go/harness/runner.go:288]: Oops, heh

File: pyproject.toml [hunk 5]
```diff
@@ -12,6 +12,7 @@ packages = [
 [tool.poetry.dependencies]
 python = "^3.7"
 temporalio = "^0.1a2"
+setuptools = "^63.3"
```
Comment (cretz) [inline pyproject.toml:15]: Can remove this now that we know it's a system dependency

File: features/query/successful_query/feature.go (hunk 3 above)
Comment (Sushisource) [reply features/query/successful_query/feature.go]: :cry:

File: features/query/successful_query/feature.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/workflow"
+)
+
+var Feature = harness.Feature{
+	Workflows: Workflow,
+	CheckResult: func(ctx context.Context, r *harness.Runner, run client.WorkflowRun) error {
+		var qRes int
+		q1, err := r.Client.QueryWorkflow(ctx, run.GetID(), run.GetRunID(), "counterQ")
+		r.Assert.NoError(err)
```
Comment (cretz) [inline features/query/successful_query/feature.go:15]: Change these `r.Assert` to `r.Require` or do the `if !r.Assert.NoError(err) { return r.LastAssertErr }` (sorry, this is just due to the ugliness of https://pkg.go.dev/github.com/stretchr/testify/assert and https://pkg.go.dev/github.com/stretchr/testify/require)

Comment (Sushisource) [reply features/query/successful_query/feature.go:15]: Thanks. For some reason it didn't seem like NoError existed in that package.. weird. Fixed.

**Review by cretz — APPROVED** [review]

**Pull request merged**